c2rust-analyze/tests/analyze/string_casts.rs

```diff
@@ -33,3 +33,57 @@ pub fn cast_from_literal() {
 pub fn cast_from_literal_explicit() {
     std::ptr::addr_of!(*b"\0") as *const u8 as *const core::ffi::c_char;
 }
+
+/// [`PointerCast::ReifyFnPointer`]
+/// Can't figure out how to create a [`PointerCast::ReifyFnPointer`].
```

Comment:
I thought that would happen here but it didn't (it turned into some other cast).  I casted `impl Fn(u8) -> i8` to `fn(u8) -> i8`, and I thought `impl Fn(u8) -> i8` could be a `FnDef`.  Let me try again with a more explicit case like your example.

Reply:
> I thought `impl Fn(u8) -> i8` could be a `FnDef`

It could be, but isn't guaranteed to be, so the MIR can't use the `FnDef`-specific `PointerCast`.

Reply:
Ahh.

Reply:
Fixed in e02903c7f11fc719062ecc13bb0ca5a0d9e2655d.